### ISSUE TITLE
Some ydl downloads not pickd up by aria2c

### ISSUE THREAD

**Issue by aidinzolghadr** [aria2/aria2#1495]
Hi,

I'm using this command to download from Youtube:

Problem is, a lot of times downloads are not picked up by aria2c and uses ydl's internal downloader which is very slow.

This is an example of my command and also a video that won't get picked up by Aria2c:

`youtube-dl.exe --external-downloader aria2c --external-downloader-args '-c -j 16 -x 16 -s 16 -k 1M' -i -f bestvideo[ext=mp4]+bestaudio[ext=m4a]/mp4 https://www.youtube.com/watchv?v=Wkgs2jCelvk`

What am I missing here?

Updated both to latest versions.

Thanks.

**Comment by Jimmy-Z** [comment]
I'm pretty sure you should fire this to youtube-dl.

And I'd assume aria2c is not in your path? I'd specify an absolute path to aria2c.exe.

**Comment by aidinzolghadr** [comment]
They told me to post it here.

aria2c is in the path in fact since some downloads are picked up by it. I was hoping some of my passed parameters causes this aria2c to skip my downloads maybe certain file sizes causes are not compatible with my parameters, etc.

Thanks.

**Comment by Jimmy-Z** [comment]
In that case, what exactly did they say? link to that issue?

**Comment by aidinzolghadr** [comment]
[https://github.com/ytdl-org/youtube-dl/issues/22134]((https://github.com/ytdl-org/youtube-dl/issues/22134))

**Comment by Jimmy-Z** [comment]
Unfortunately not much info there, could you get aria2's log when it's invoked by ydl?

**Comment by tjesp** [comment]
Ok, shortly after I wrote my response I found my own answer. I am pretty new on Mac OS so pardon me. Mac OS was recently updated and I needed to follow some steps to get my tools working fully in the terminal after an OS update:

https://apple.stackexchange.com/questions/359821/whats-the-best-way-to-update-homebrew-when-upgrading-macos

I did all steps outlined in all answers and now things seem to work. If the original poster is also using Mac OS he might want to try as well.

**Comment by Jimmy-Z** [comment]
OP is using a "aria2.exe" so I'd guess your solution won't apply to him.

**Comment by aidinzolghadr** [comment]
Yep, using Windows 10's PowerShell. Also I don't get an error, it just uses ydl's internal downloader and seems to skip Aria2c.

**Comment by coding-function** [comment]
I had stumble upon the same issue but i doubt it has anything to do with aria2 because i was trying to download videos from a playlist which has 45 videos in it and for the first video aria2 was used as downloading client and for the next few yt-dl used its own default downloader. it seems some how it depends upon the extractor in yt-dl that decides whether to pass downloading to external downloader or not and some downloads are passed to ffmpeg download even if you ask them to download with aria2.

**Comment by aidinzolghadr** [comment]
I'm usually using it as a single call, like the one in original post but they don't get picked up by aria2c. I was hoping that some of my switches are causing this, somehow not compatible with some videos like for example size of the video is in a certain criteria that can't be chopped to 16 or something.

**Comment by Jimmy-Z** [comment]
There has to be some way to expose aria2's log output, then, you can determine what's wrong from the log.

If ydl doesn't provide an option to do that(I doubt that though, do dig into their man page), you can use aria2's option to write log to a file.

If you suspect "can't be chopped to 16" is the culprit, then try removing that part and see if it works then, trial and error is an important part of debugging.

If aria2 is not called at all, that's ydl's bug.

**Comment by aidinzolghadr** [comment]
If I remove all the external args from aria2c, the video in the first post still won't get picked up by aria2c so I think parameters being the problem is not the case here.

Can someone please try [this video](https://www.youtube.com/watch?v=Wkgs2jCelvk) and see if it gets picked up by aria2c and if yes post the exact command so I can test as well? Maybe something is with my machine.

**Comment by Anon-Exploiter** [comment]
@aidinzolghadr same issue here, did you resolve it?
